src/main/kotlin/com/exactpro/th2/check1/rule/RuleFactory.kt

```diff
@@ -0,0 +1,198 @@
+                            filter,
+                            request.parentEventId,
+                            streamObservable,
+                            eventBatchRouter
+                    )
+                }
+                onErrorEvent(request.parentEventId) {
+                    Event.start()
+                            .name("Check rule cannot be created")
+                            .type("checkRuleCreation")
+                }
+            }
+
+    fun createSequenceCheckRule(request: CheckSequenceRuleRequest): SequenceCheckRuleTask =
+            ruleCreation(request) {
+                checkAndCreateRule { request ->
+                    check(request.hasParentEventId()) { "Parent event id can't be null" }
+                    check(request.connectivityId.sessionAlias.isNotEmpty()) { "Session alias cannot be empty" }
+                    val sessionAlias: String = request.connectivityId.sessionAlias
+                    val direction = directionOrDefault(request.direction)
+
+                    check((request.messageFiltersList.isEmpty() && request.rootMessageFiltersList.isNotEmpty())
+                            || (request.messageFiltersList.isNotEmpty() && request.rootMessageFiltersList.isEmpty())) {
+                        "Either messageFilters or rootMessageFilters must be set but not both"
+                    }
+
+                    val protoMessageFilters: List<RootMessageFilter> = request.rootMessageFiltersList.ifEmpty {
+                        request.messageFiltersList.map { it.toRootMessageFilter() }
+                    }
+
+                    SequenceCheckRuleTask(
+                            request.description,
+                            Instant.now(),
+                            SessionKey(sessionAlias, direction),
+                            request.timeout,
+                            maxEventBatchContentSize,
+                            request.preFilter,
+                            protoMessageFilters,
+                            request.checkOrder,
+                            request.parentEventId,
+                            streamObservable,
+                            eventBatchRouter
+                    )
+                }
+                onErrorEvent(request.parentEventId) {
+                    Event.start()
+                            .name("Sequence check rule cannot be created")
+                            .type("sequenceCheckRuleCreation")
+                }
+            }
+
+
+    private fun <T : GeneratedMessageV3, R : AbstractCheckTask> ruleCreation(request: T, block: RuleCreationContext<T, R>.() -> Unit): R {
+        val ruleCreationContext = RuleCreationContext<T, R>().apply(block)
+        try {
+            return ruleCreationContext.action(request)
+        } catch (e: RuleInternalException) {
+            throw e
+        } catch (e: Exception) {
+            val rootEvent = ruleCreationContext.event()
+            rootEvent.addSubEventWithSamePeriod()
```

Comment:
Please, add the exception to the event. There is a method in Event class that allows you to do that